core/src/main/java/io/grpc/CallOptions.java

```diff
@@ -370,7 +371,7 @@ public String toString() {
     toStringHelper.add("affinity", affinity);
     toStringHelper.add("executor", executor != null ? executor.getClass() : null);
     toStringHelper.add("compressorName", compressorName);
-    toStringHelper.add("customOptions", Arrays.toString(customOptions));
+    toStringHelper.add("customOptions", Arrays.deepToString(customOptions));
```

Comment:
If we try `CallOptions.Default.withOption(option1, "v1").withOption(option1,"v2").toString()`,
it outputs
`... customOptions=[[option1, v2], [option1, v1]], ...` 
Should we re-implement `withOption()` so that `toString()` can look nicer?